pagefaultgames/pokerogue
# [Move] Add makeContact To Physical Part Of Shell Side Arm

(#3503)


## What are the changes the user will see?

The user now makes physical contact if Shell Side Arm does physical damage.


## What are the changes from a developer perspective?

Now calls `makesContact()` if move turns physical.

Removed `.partial()` 

Added test that checks for contact

## Screenshots/Videos

Before changes vs Rough Skin:

[shell-side-arm-implementation-before.webm](https://github.com/user-attachments/assets/b858a4ad-c51f-48c2-ae12-5e86841b4739)

After changes vs Rough Skin:

[shell-side-arm-implementation-after.webm](https://github.com/user-attachments/assets/13967dd3-e747-4091-a9ac-34aac31340a7)

## How to test the changes?

Give enemy Pokemon Rough Skin (Ability) and use Side Shell Arm. Make sure the enemy Pokemon has more SPDEF than DEF.

```
const overrides = {
  OPP_SPECIES_OVERRIDE: Species.SNORLAX,
  OPP_MOVESET_OVERRIDE: [ Moves.SPLASH ],
  OPP_ABILITY_OVERRIDE: Abilities.ROUGH_SKIN,
  OPP_LEVEL_OVERRIDE: 20,
  MOVESET_OVERRIDE: [ Moves.SHELL_SIDE_ARM ],
} satisfies Partial<InstanceType<typeof DefaultOverrides>>;
```

## Checklist
